nixos/modules/system/boot/networkd.nix

```diff
@@ -1911,12 +1907,13 @@ in
 
   };
 
-  config = mkMerge [
+  commonConfig = config: let cfg = config.systemd.network; in mkMerge [
```

Comment:
Normally does are defined at the very top but since the file is already very big I am a bit unsure here.

Reply:
@SuperSandro2000 `commonConfig` is an abstraction over the same options in both stage 1 and stage 2. They need different configs, so it has to be a function argument.